pkg/adapter/maintenance/mode.go

```diff
@@ -1,82 +1,58 @@
 package maintenance
 
 import (
-	"github.com/cloudogu/cesapp-lib/registry"
-
+	"context"
+	"fmt"
 	"github.com/cloudogu/k8s-blueprint-operator/pkg/domainservice"
+	liberrors "github.com/cloudogu/k8s-registry-lib/errors"
+	"github.com/cloudogu/k8s-registry-lib/repository"
 )
 
 // Mode contains methods to Activate and Deactivate the switcher.
 // When it is active, a page is displayed to the user, telling them that there is maintenance going on.
 type Mode struct {
-	lock
-	switcher
+	libAdapter libMaintenanceModeAdapter
 }
 
-func New(globalConfig registry.ConfigurationContext) *Mode {
+func NewMaintenanceModeAdapter(libAdapter libMaintenanceModeAdapter) *Mode {
 	return &Mode{
-		lock:     &defaultLock{globalConfig: globalConfig},
-		switcher: &defaultSwitcher{globalConfig: globalConfig},
+		libAdapter: libAdapter,
 	}
 }
 
 // Activate enables the maintenance mode, setting the given MaintenancePageModel
-func (m *Mode) Activate(content domainservice.MaintenancePageModel) error {
-	isActive, isOurs, err := m.lock.isActiveAndOurs()
-	if err != nil {
-		return &domainservice.InternalError{
-			WrappedError: err,
-			Message:      "failed to check if maintenance mode is already active and ours",
-		}
-	}
-
-	if isActive && !isOurs {
-		return &domainservice.ConflictError{
-			WrappedError: nil,
-			Message:      "cannot activate maintenance mode as it was already activated by another party",
-		}
-	}
-
-	err = m.switcher.activate(content)
+func (m *Mode) Activate(ctx context.Context, title, text string) error {
+	err := m.libAdapter.Activate(ctx, repository.MaintenanceModeDescription{
+		Title: title,
+		Text:  text,
+	})
+	err = mapToBlueprintError(err)
 	if err != nil {
-		return &domainservice.InternalError{
-			WrappedError: err,
-			Message:      "failed to activate maintenance mode",
-		}
+		return fmt.Errorf("could not activate maintenance mode: %w", err)
 	}
-
 	return nil
 }
 
 // Deactivate disables the maintenance mode.
-func (m *Mode) Deactivate() error {
-	isActive, isOurs, err := m.lock.isActiveAndOurs()
+func (m *Mode) Deactivate(ctx context.Context) error {
+	err := m.libAdapter.Deactivate(ctx)
+	err = mapToBlueprintError(err)
 	if err != nil {
-		return &domainservice.InternalError{
-			WrappedError: err,
-			Message:      "failed to check if maintenance mode is already active and ours",
-		}
-	}
-
-	if !isActive {
-		// do nothing
-		return nil
-	}
-
-	if !isOurs {
-		return &domainservice.ConflictError{
-			WrappedError: nil,
-			Message:      "cannot deactivate maintenance mode as it was activated by another party",
-		}
+		return fmt.Errorf("could not activate maintenance mode: %w", err)
 	}
+	return nil
+}
 
-	err = m.switcher.deactivate()
+func mapToBlueprintError(err error) error {
```

Comment:
If we further want to map the errors, then we need to do that in the adapters package as no implementation details of the adapters should leak into the application/domain. We can create a reusable function there too. My concern was, that i wanted fitting error messages. 

an idea for the refactoring is, that we remove our own error types and only use these common error types in the domain. This would lead to less abstraction and a dependency to the k8s-registry-lib (or later to ces-commons where the errors should be instead) but it would remove complexity while using our own repositories and adapters which are abstractions of the real implementations themself.

Reply:
I thought that we could pass in part of the error message. 
so that you have something like `("could not %s due to conflict error", action)` in the map-method.

But using the errors from the lib directly is a good idea, I think :)